recipes-core/searcher-ssh-key/files/ns-enter.c

```diff
@@ -0,0 +1,90 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <sched.h>
+
+#define MAX_RETRIES 5
+#define RETRY_DELAY 1
+#define SEARCHER_UID 1000
+#define SEARCHER_GID 1000
+
+int wait_for_namespace() {
+    int retries = 0;
+    while (retries < MAX_RETRIES) {
+        if (access("/var/run/netns/searcher-ns", F_OK) == 0) {
+            return 1;
+        }
+        sleep(RETRY_DELAY);
+        retries++;
+    }
+    return 0;
+}
+
+int enter_namespace() {
```

Comment:
what are the restrictions of this namespace? is it only network wise?

Reply:
This is all described in details in the searcher-net-init script below.